librz/bin/p/bin_elf.c

```diff
@@ -159,6 +159,9 @@ RzBinPlugin rz_bin_plugin_elf = {
 	.file_type = &get_file_type,
 	.regstate = &regstate,
 	.maps = &maps,
+	.section_type_to_string = &Elf_(section_type_to_string),
+	//.section_flag_to_string = &Elf_(section_flag_to_string),
```

Comment:
Just remove the commented ode.

Reply:
There is a lot of commented code that I have to cleanup. I will do them in my next commit. 

> Overall the output looks better, but some previously working files still return the error:
> 
> ```
> [XX] db/formats/web_assembly WASM: Wasm - sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/wasm/binary.wasm
> -- stdout
> @@ -1,9 +0,0 @@
> -[Sections]
> -
> -nth paddr        size vaddr       vsize perm name
> --------------------------------------------------
> -0   0x0000000a   0x10 0x0000000a   0x10 ---- type
> -1   0x0000001d   0x2c 0x0000001d   0x2c ---- function
> -2   0x0000004d  0x1dc 0x0000004d  0x1dc ---- export
> -3   0x0000022d  0x1f6 0x0000022d  0x1f6 ---- code
> -
> 
> 
> -- stderr
> WARNING: No calling convention defined for this file, analysis may be inaccurate.
> [XX] db/formats/qnx sections of qnx
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/qnx/test.x
> -- stdout
> @@ -1,7 +0,0 @@
> -[Sections]
> -
> -nth paddr        size vaddr       vsize perm name
> --------------------------------------------------
> -0   0x00000044   0xf6 0x00000044   0xf6 ---- LMF_LOAD
> -1   0x00000152  0x6c3 0x00000152  0x6c3 ---- LMF_LOAD
> -
> 
> [XX] db/formats/omf sections 32 bits omf
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/omf/hello_world32
> -- stdout
> 
> -- exit status: 
> [XX] db/formats/omf sections 16 bits omf
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/omf/hello_world
> -- stdout
> 
> -- exit status: 
> [XX] db/formats/nso nso sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -e io.cache=true -Qc iS bins/nso/application.nso
> -- stdout
> @@ -1,9 +0,0 @@
> -[Sections]
> -
> -nth paddr          size vaddr         vsize perm name
> ------------------------------------------------------
> -0   0x00000000    0x100 0x00000000    0x100 -r-- header
> -1   0x00000100  0x20e18 0x08000000  0x20e18 -r-x text
> -2   0x0001718f   0xdb0c 0x08021000   0xdb0c -r-- ro
> -3   0x0001f856   0x4218 0x0802f000   0x4218 -rw- data
> -
> 
> 
> -- stderr
> load_bytes in bin.nso must die
> MOD Offset = 0x130
> magic 30444f4d at 0x130
> is mode0
> magic 0x30444f4d
> dynamic 0x2c1b8
> bss 0x330e8 0x3ff48
> unwind 0x26f38 0x26f38
> -------------
> mod 0x130
> next 0x2c1b830444f4d
> prev 0x3ff48000330e8
> base 0xa9bf73fba9bf7bfd
> init 0xa8c16bf9a8c163f7
> fini 0xa8c17bfda8c173fb
> relplt 0x5642fb3c5642f33b
> symtab = 0x56408c0456408403
> strtab = 0x56408c0456408403
> strtabsz = 0x0
> BSS Size 0x0000ce60
> 
> -- exit status: 
> [XX] db/formats/nro nro sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -e io.cache=true -Qc iS bins/nro/appstore.nro
> -- stdout
> @@ -1,10 +0,0 @@
> -[Sections]
> -
> -nth paddr           size vaddr          vsize perm name
> --------------------------------------------------------
> -0   0x00000000      0x80 0x00000000      0x80 -r-- header
> -1   0x0021d000       0x0 0x0021d000       0x0 -r-- sig0
> -2   0x00000000  0x1b4000 0x00000000  0x1b4000 -r-x text
> -3   0x001b4000   0x5a000 0x001b4000   0x5a000 -r-- ro
> -4   0x0020e000    0xf000 0x0020e000    0xf000 -rw- data
> -
> 
> 
> -- stderr
> magic 14000020 at 0x0
> Invalid MOD0 address
> Base Address 0x00000000
> BSS Size 0x0001e000
> 
> -- exit status: 
> [XX] db/formats/ne NE Segments
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iSS bins/ne/anim8.exe
> -- stdout
> @@ -1,7 +0,0 @@
> -[Segments]
> -
> -nth paddr         size vaddr        vsize perm name
> ----------------------------------------------------
> -0   0x00000a00  0x54ee 0x00000a00  0x54ee ---x MOVEABLE.2560
> -1   0x00007600  0x3ec4 0x00007600  0x3ec4 -rwx MOVEABLE.30208
> -
> 
> 
> -- exit status: 
> [XX] db/formats/mdmp : sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS~calc bins/mdmp/calc.dmp
> -- stdout
> @@ -1 +1 @@
> -9   0x00000000   0xe3000 0xfffe0000      0xe3000 ---- C:_Windows_System32_calc.exe
> +9   0x00000000   0xe3000 0xfffe0000      0xe3000 ---- C:_Windows_System32_calc.exe          
> 
> [XX] db/formats/mach0/mach0 mach0 segments
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iSS bins/mach0/hello-osx-x86_64
> -- stdout
> @@ -1,7 +0,0 @@
> -[Segments]
> -
> -nth paddr         size vaddr         vsize perm name
> -――――――――――――――――――――――――――――――――――――――――――――――――――――
> -0   0x00000000  0x1000 0x100000000  0x1000 -r-x TEXT
> -1   0x00001000  0x1000 0x100001000  0x1000 -rw- DATA
> -2   0x00002000  0x1000 0x100002000  0x1000 -r-- LINKEDIT
> 
> 
> -- exit status: 
> [XX] db/formats/mach0/mach0 mach0 sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/mach0/hello-osx-x86_64
> -- stdout
> @@ -1,14 +0,0 @@
> -[Sections]
> -
> -nth paddr       size vaddr        vsize perm name                     type                     flags
> -----------------------------------------------------------------------------------------------------
> -0   0x00000ed0  0x53 0x100000ed0   0x53 -r-x 0.__TEXT.__text          REGULAR                  SOME_INSTRUCTIONS,PURE_INSTRUCTIONS
> -1   0x00000f24   0xc 0x100000f24    0xc -r-x 1.__TEXT.__symbol_stub1  SYMBOL_STUBS             SOME_INSTRUCTIONS,PURE_INSTRUCTIONS
> -2   0x00000f30   0xc 0x100000f30    0xc -r-x 2.__TEXT.__cstring       CSTRING_LITERALS               
> -3   0x00000f3c  0x24 0x100000f3c   0x24 -r-x 3.__TEXT.__stub_helper   REGULAR                  SOME_INSTRUCTIONS,PURE_INSTRUCTIONS
> -4   0x00000f60  0x54 0x100000f60   0x54 -r-x 4.__TEXT.__unwind_info   REGULAR                        
> -5   0x00000fb8  0x48 0x100000fb8   0x48 -r-x 5.__TEXT.__eh_frame      COALESCED                STRIP_STATIC_SYMS,NO_TOC
> -6   0x00001000  0x28 0x100001000   0x28 -rw- 6.__DATA.__program_vars  REGULAR                        
> -7   0x00001028  0x10 0x100001028   0x10 -rw- 7.__DATA.__nl_symbol_ptr NON_LAZY_SYMBOL_POINTERS       
> -8   0x00001038  0x10 0x100001038   0x10 -rw- 8.__DATA.__la_symbol_ptr LAZY_SYMBOL_POINTERS           
> -9   0x00001048  0x20 0x100001048   0x20 -rw- 9.__DATA.__data          REGULAR                        
> 
> 
> -- exit status: 
> [XX] db/formats/le GNUGREP.DLL sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/le/GNUGREP.DLL
> -- stdout
> @@ -1,16 +0,0 @@
> -[Sections]
> -
> -nth paddr         size vaddr        vsize perm name
> ----------------------------------------------------
> -0   0x00001600  0x1000 0x00010000  0x1000 -r-x obj.1.page.0
> -1   0x00002600  0x1000 0x00011000  0x1000 -r-x obj.1.page.1
> -2   0x00003600  0x1000 0x00012000  0x1000 -r-x obj.1.page.2
> -3   0x00004600  0x1000 0x00013000  0x1000 -r-x obj.1.page.3
> -4   0x00005600  0x1000 0x00014000  0x1000 -r-x obj.1.page.4
> -5   0x00006600  0x1000 0x00015000  0x1000 -r-x obj.1.page.5
> -6   0x00007600  0x1000 0x00016000  0x1000 -r-x obj.1.page.6
> -7   0x00008600   0xe00 0x00017000  0x1000 -r-x obj.1.page.7
> -8   0x00009400   0x200 0x00020000  0x1000 -r-x obj.2.page.0
> -9   0x00009600   0x200 0x00030000  0x1000 -r-x obj.3.page.0
> -10  0x00001400   0x200 0x00040000  0x1000 -rw- obj.4.page.0
> -
> 
> 
> -- exit status: 
> [XX] db/formats/le GCC.EXE sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/le/GCC.EXE
> -- stdout
> @@ -1,23 +0,0 @@
> -[Sections]
> -
> -nth paddr         size vaddr           vsize perm name
> -------------------------------------------------------
> -0   0x00001000  0x1000 0x00010000     0x1000 -r-x obj.1.page.0
> -1   0x00002000  0x1000 0x00011000     0x1000 -r-x obj.1.page.1
> -2   0x00003000  0x1000 0x00012000     0x1000 -r-x obj.1.page.2
> -3   0x00004000  0x1000 0x00013000     0x1000 -r-x obj.1.page.3
> -4   0x00005000  0x1000 0x00014000     0x1000 -r-x obj.1.page.4
> -5   0x00006000  0x1000 0x00015000     0x1000 -r-x obj.1.page.5
> -6   0x00007000  0x1000 0x00016000     0x1000 -r-x obj.1.page.6
> -7   0x00008000  0x1000 0x00017000     0x1000 -r-x obj.1.page.7
> -8   0x00009000  0x1000 0x00018000     0x1000 -r-x obj.1.page.8
> -9   0x0000a000  0x1000 0x00019000     0x1000 -r-x obj.1.page.9
> -10  0x0000b000  0x1000 0x0001a000     0x1000 -r-x obj.1.page.10
> -11  0x0000c000  0x1000 0x0001b000     0x1000 -r-x obj.1.page.11
> -12  0x0000d000  0x1000 0x0001c000     0x1000 -r-x obj.1.page.12
> -13  0x0000e000  0x1000 0x0001d000     0x1000 -r-x obj.1.page.13
> -14  0x0000f000  0x1000 0x0001e000     0x1000 -r-x obj.1.page.14
> -15  0x00010000  0x1000 0x00020000     0x1000 -rw- obj.2.page.0
> -16  0x00000000     0x0 0x00030000  0x2000000 -rw- obj.3
> -17  0x00000000     0x0 0x02030000   0x800000 -rw- obj.4
> -
> 
> 
> -- exit status: 
> [XX] db/formats/le cdogs.exe sections
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iS bins/le/cdogs.exe
> -- stdout
> @@ -1,66 +0,0 @@
> -[Sections]
> -
> -nth paddr         size vaddr        vsize perm name
> ----------------------------------------------------
> -0   0x0000d200  0x1000 0x00010000  0x1000 -r-x obj.1.page.0
> -1   0x0000e200  0x1000 0x00011000  0x1000 -r-x obj.1.page.1
> -2   0x0000f200  0x1000 0x00012000  0x1000 -r-x obj.1.page.2
> -3   0x00010200  0x1000 0x00013000  0x1000 -r-x obj.1.page.3
> -4   0x00011200  0x1000 0x00014000  0x1000 -r-x obj.1.page.4
> -5   0x00012200  0x1000 0x00015000  0x1000 -r-x obj.1.page.5
> -6   0x00013200  0x1000 0x00016000  0x1000 -r-x obj.1.page.6
> -7   0x00014200  0x1000 0x00017000  0x1000 -r-x obj.1.page.7
> -8   0x00015200  0x1000 0x00018000  0x1000 -r-x obj.1.page.8
> -9   0x00016200  0x1000 0x00019000  0x1000 -r-x obj.1.page.9
> -10  0x00017200  0x1000 0x0001a000  0x1000 -r-x obj.1.page.10
> -11  0x00018200  0x1000 0x0001b000  0x1000 -r-x obj.1.page.11
> -12  0x00019200  0x1000 0x0001c000  0x1000 -r-x obj.1.page.12
> -13  0x0001a200  0x1000 0x0001d000  0x1000 -r-x obj.1.page.13
> -14  0x0001b200  0x1000 0x0001e000  0x1000 -r-x obj.1.page.14
> -15  0x0001c200  0x1000 0x0001f000  0x1000 -r-x obj.1.page.15
> -16  0x0001d200  0x1000 0x00020000  0x1000 -r-x obj.1.page.16
> -17  0x0001e200  0x1000 0x00021000  0x1000 -r-x obj.1.page.17
> -18  0x0001f200  0x1000 0x00022000  0x1000 -r-x obj.1.page.18
> -19  0x00020200  0x1000 0x00023000  0x1000 -r-x obj.1.page.19
> -20  0x00021200  0x1000 0x00024000  0x1000 -r-x obj.1.page.20
> -21  0x00022200  0x1000 0x00025000  0x1000 -r-x obj.1.page.21
> -22  0x00023200  0x1000 0x00026000  0x1000 -r-x obj.1.page.22
> -23  0x00024200  0x1000 0x00027000  0x1000 -r-x obj.1.page.23
> -24  0x00025200  0x1000 0x00028000  0x1000 -r-x obj.1.page.24
> -25  0x00026200  0x1000 0x00029000  0x1000 -r-x obj.1.page.25
> -26  0x00027200  0x1000 0x0002a000  0x1000 -r-x obj.1.page.26
> -27  0x00028200  0x1000 0x0002b000  0x1000 -r-x obj.1.page.27
> -28  0x00029200  0x1000 0x0002c000  0x1000 -r-x obj.1.page.28
> -29  0x0002a200  0x1000 0x0002d000  0x1000 -r-x obj.1.page.29
> -30  0x0002b200  0x1000 0x0002e000  0x1000 -r-x obj.1.page.30
> -31  0x0002c200  0x1000 0x0002f000  0x1000 -r-x obj.1.page.31
> -32  0x0002d200  0x1000 0x00030000  0x1000 -r-x obj.1.page.32
> -33  0x0002e200  0x1000 0x00031000  0x1000 -r-x obj.1.page.33
> -34  0x0002f200  0x1000 0x00032000  0x1000 -r-x obj.1.page.34
> -35  0x00030200  0x1000 0x00033000  0x1000 -r-x obj.1.page.35
> -36  0x00031200  0x1000 0x00034000  0x1000 -r-x obj.1.page.36
> -37  0x00032200  0x1000 0x00035000  0x1000 -r-x obj.1.page.37
> -38  0x00033200  0x1000 0x00036000  0x1000 -r-x obj.1.page.38
> -39  0x00034200  0x1000 0x00037000  0x1000 -r-x obj.1.page.39
> -40  0x00035200  0x1000 0x00038000  0x1000 -r-x obj.1.page.40
> -41  0x00036200  0x1000 0x00039000  0x1000 -r-x obj.1.page.41
> -42  0x00037200  0x1000 0x0003a000  0x1000 -r-x obj.1.page.42
> -43  0x00038200  0x1000 0x0003b000  0x1000 -r-x obj.1.page.43
> -44  0x00039200  0x1000 0x0003c000  0x1000 -r-x obj.1.page.44
> -45  0x0003a200  0x1000 0x00040000  0x1000 -rw- obj.2.page.0
> -46  0x0003b200  0x1000 0x00041000  0x1000 -rw- obj.2.page.1
> -47  0x0003c200  0x1000 0x00042000  0x1000 -rw- obj.2.page.2
> -48  0x0003d200  0x1000 0x00043000  0x1000 -rw- obj.2.page.3
> -49  0x0003e200  0x1000 0x00044000  0x1000 -rw- obj.2.page.4
> -50  0x0003f200  0x1000 0x00045000  0x1000 -rw- obj.2.page.5
> -51  0x00040200  0x1000 0x00046000  0x1000 -rw- obj.2.page.6
> -52  0x00041200  0x1000 0x00047000  0x1000 -rw- obj.2.page.7
> -53  0x00042200  0x1000 0x00048000  0x1000 -rw- obj.2.page.8
> -54  0x00043200  0x1000 0x00049000  0x1000 -rw- obj.2.page.9
> -55  0x00044200  0x1000 0x0004a000  0x1000 -rw- obj.2.page.10
> -56  0x00045200  0x1000 0x0004b000  0x1000 -rw- obj.2.page.11
> -57  0x00046200  0x1000 0x0004c000  0x1000 -rw- obj.2.page.12
> -58  0x00047200  0x1000 0x0004d000  0x1000 -rw- obj.2.page.13
> -59  0x00048200  0x1000 0x0004e000  0x1000 -rw- obj.2.page.14
> -60  0x00049200   0xd8c 0x0004f000  0x1000 -rw- obj.2.page.15
> -
> ```
> 
> Also segments output (`iSS`) still broken:
> 
> ```
> [XX] db/cmd/cmd_i iSS (file x86_64)
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iSS bins/elf/analysis/hello-linux-x86_64
> -- stdout
> @@ -1,14 +0,0 @@
> -[Segments]
> -
> -nth paddr        size vaddr       vsize perm name
> --------------------------------------------------
> -0   0x00000040  0x1c0 0x00400040  0x1c0 -r-x PHDR
> -1   0x00000200   0x1a 0x00400200   0x1a -r-- INTERP
> -2   0x00000000  0x6dc 0x00400000  0x6dc -r-x LOAD0
> -3   0x000006e0  0x230 0x006006e0  0x238 -rw- LOAD1
> -4   0x000006f8  0x1d0 0x006006f8  0x1d0 -rw- DYNAMIC
> -5   0x0000021c   0x44 0x0040021c   0x44 -r-- NOTE
> -6   0x000005d0   0x34 0x004005d0   0x34 -r-- GNU_EH_FRAME
> -7   0x00000000    0x0 0x00000000    0x0 -rw- GNU_STACK
> -8   0x00000000   0x40 0x00400000   0x40 -rw- ehdr
> -
> 
> 
> -- exit status: 
> [XX] db/cmd/cmd_i iS (file x86_64)
> ```
> 
> https://builds.sr.ht/~xvilka/job/455519#task-test-1326

I will try to fix these but I can't really tell why these aren't working. Could you maybe suggest what you think is happening that it doesn't give out the dump?
I tried these on different binaries locally and it seems to work but not here.

Reply:
I think its due to the rz_table format string having the wrong number of format specifiers. I will have a commit that should clean up the segment `iSS` output soon